warehouse/query-core/src/test/java/datawave/query/jexl/visitors/FieldToFieldComparisonVisitorTest.java

```diff
@@ -3,39 +3,102 @@
+        
+        assertResult("FOO < 1", "FOO < 1");
+        assertResult("(FOO || BAR).min().hashCode() < 0", "(FOO || BAR).min().hashCode() < 0");
     }
     
     @Test
-    public void testEqDoNothingFieldsToLiteral() throws ParseException {
-        ASTJexlScript query = FieldToFieldComparisonVisitor.forceEvaluationOnly(JexlASTHelper.parseJexlQuery("(FOO || BAR).min().hashCode() == 0"));
-        assertFalse(ASTEvaluationOnly.instanceOf(query));
+    public void testLE() throws ParseException {
+        assertResult("FOO <= BAR", "((_Eval_ = true) && (FOO <= BAR))");
+        
+        assertResult("FOO <= 1", "FOO <= 1");
+        assertResult("(FOO || BAR).min().hashCode() <= 0", "(FOO || BAR).min().hashCode() <= 0");
     }
     
     @Test
-    public void testEqDoNothing2() throws ParseException {
-        ASTJexlScript query = FieldToFieldComparisonVisitor.forceEvaluationOnly(JexlASTHelper.parseJexlQuery("(UUID =~ 'C.*?' || UUID =~ 'S.*?')"));
-        assertFalse(ASTEvaluationOnly.instanceOf(query));
+    public void testGT() throws ParseException {
+        assertResult("FOO > BAR", "((_Eval_ = true) && (FOO > BAR))");
+        
+        assertResult("FOO > 1", "FOO > 1");
+        assertResult("(FOO || BAR).min().hashCode() > 0", "(FOO || BAR).min().hashCode() > 0");
+    }
+    
+    @Test
+    public void testGE() throws ParseException {
+        assertResult("FOO >= BAR", "((_Eval_ = true) && (FOO >= BAR))");
+        
+        assertResult("FOO >= 1", "FOO >= 1");
+        assertResult("(FOO || BAR).min().hashCode() >= 0", "(FOO || BAR).min().hashCode() >= 0");
+    }
+    
+    @Test
+    public void testRegexOpsAreNotModified() throws ParseException {
+        assertResult("(UUID =~ 'C.*?' || UUID =~ 'S.*?')", "(UUID =~ 'C.*?' || UUID =~ 'S.*?')");
+        assertResult("(UUID !~ 'C.*?' || UUID !~ 'S.*?')", "(UUID !~ 'C.*?' || UUID !~ 'S.*?')");
     }
     
     @Test(expected = ParseException.class)
-    public void testEqDoNotSupport() throws ParseException {
+    public void testChainedEQThrowsException() throws ParseException {
         JexlASTHelper.parseJexlQuery("FIELD_A == FIELD_B == FIELD_C");
     }
+    
+    private void assertResult(String original, String expected) throws ParseException {
+        ASTJexlScript originalScript = JexlASTHelper.parseJexlQuery(original);
+        
+        ASTJexlScript visitedScript = FieldToFieldComparisonVisitor.forceEvaluationOnly(originalScript);
+        
+        // Verify the script is as expected, and has a valid lineage.
+        assertScriptEquality(visitedScript, expected);
+        assertLineage(visitedScript);
+        
+        // Verify the original script was not modified, and still has a valid lineage.
+        assertScriptEquality(originalScript, original);
+        assertLineage(originalScript);
+    }
+    
+    private void assertScriptEquality(ASTJexlScript actualScript, String expected) throws ParseException {
```

Comment:
These two functions (this and the one below) seem like generalized utilities for asserting queries are equal with valid lineage. This would be a good candidate some kind of test utility that could be reused across all of these lineage verification/fix PRs.